components/navigation/BottomBar.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { View, StyleSheet, Dimensions } from "react-native";
+import { ThemedIconButton } from "@/components/theme/ThemedIconButton";
+import { Colors } from "@/constants/Colors";
+import { useThemeColor } from "@/hooks/useThemeColor";
+
+// Get screen width and height
+const { width, height } = Dimensions.get("window");
+
+interface BottomBarProps {
+  leftIcon?: string;
+  leftAction?: () => void;
+  centerIcon?: string;
+  centerAction?: () => void;
+  rightIcon?: string;
+  rightAction?: () => void;
+  colorType?: keyof typeof Colors.light & keyof typeof Colors.dark;
```

Comment:
Perhaps this should be of type `Colors` directly. I understand what's used there but I believe it leads to less readable code.
This would imply changing the function call everywhere where this is used, so perhaps keeping it the way it is is okay.

Reply:
Actually this wouldn't work as we also have lightColor and darkColor properties which we do not use yet but when we'll head into having an esthetic app will probably come in handy instead of creating a bunch of _special_ colors inside **Colors.tsx** that will only be used once.